runner/sidecar/sources.go

```diff
@@ -69,6 +69,10 @@ func connectSources(ctx context.Context, toMain func(context.Context, []byte) er
 						case dfv1.RetryNever:
 							return true, err
 						default:
+							withLock(func() {
+								step.Status.SinkStatues.IncrRetryCount(sourceName, replica)
+								newSourceMetrics(source, sourceName)
```

Comment:
delete this line, no need to register the prometheus metrics every time we do a retry